pipelineが追い付いていなくてしばらくdockerを使っていなかったのですが、一度dockerは作っていあるし立ち上がっているようなので

./docker/setDotEnv.sh
./docker-build
docker-compose up -d

は不要

docker exec -it jasmine /bin/bash
pytest

で良いのですよね？以下のエラーが出ます。操作が間違っていれば、教えてください。setup.py installもやってみましたが、結果は同じです。

root@e9ffa5ee836d:/opt/jasmine# pytest
================================================= test session starts ==================================================
platform linux -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /opt/jasmine
plugins: asdf-2.12.0
collected 28 items / 10 errors

======================================================== ERRORS ========================================================
_________________________________ ERROR collecting slowtests/test_slow_test_example.py _________________________________
ImportError while importing test module '/opt/jasmine/slowtests/test_slow_test_example.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
slowtests/test_slow_test_example.py:3: in <module>
    from test_tools.assert_timeout import assert_timeout
E   ModuleNotFoundError: No module named 'test_tools'
____________________________ ERROR collecting tests/pipeline/test_astrometric_catalogue.py _____________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_astrometric_catalogue.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_astrometric_catalogue.py:8: in <module>
    from test_pipeline import get_tests_file_name
tests/pipeline/test_pipeline.py:4: in <module>
    from telescope_baseline.tools.pipeline.analysis import Analysis
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/analysis.py:6: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
________________________________ ERROR collecting tests/pipeline/test_detector_image.py ________________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_detector_image.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_detector_image.py:3: in <module>
    from test_pipeline import get_tests_file_name
tests/pipeline/test_pipeline.py:4: in <module>
    from telescope_baseline.tools.pipeline.analysis import Analysis
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/analysis.py:6: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
____________________________ ERROR collecting tests/pipeline/test_detector_image_builder.py ____________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_detector_image_builder.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_detector_image_builder.py:8: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
___________________________ ERROR collecting tests/pipeline/test_detector_image_catalogue.py ___________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_detector_image_catalogue.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_detector_image_catalogue.py:5: in <module>
    from test_pipeline import get_tests_file_name
tests/pipeline/test_pipeline.py:4: in <module>
    from telescope_baseline.tools.pipeline.analysis import Analysis
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/analysis.py:6: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
_______________________________ ERROR collecting tests/pipeline/test_map_on_detector.py ________________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_map_on_detector.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_map_on_detector.py:10: in <module>
    from test_pipeline import get_tests_file_name
tests/pipeline/test_pipeline.py:4: in <module>
    from telescope_baseline.tools.pipeline.analysis import Analysis
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/analysis.py:6: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
___________________________ ERROR collecting tests/pipeline/test_map_on_detector_builder.py ____________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_map_on_detector_builder.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_map_on_detector_builder.py:14: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
________________________________ ERROR collecting tests/pipeline/test_map_on_the_sky.py ________________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_map_on_the_sky.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_map_on_the_sky.py:8: in <module>
    from test_pipeline import get_tests_file_name
tests/pipeline/test_pipeline.py:4: in <module>
    from telescope_baseline.tools.pipeline.analysis import Analysis
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/analysis.py:6: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
___________________________________ ERROR collecting tests/pipeline/test_pipeline.py ___________________________________
ImportError while importing test module '/opt/jasmine/tests/pipeline/test_pipeline.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/pipeline/test_pipeline.py:4: in <module>
    from telescope_baseline.tools.pipeline.analysis import Analysis
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/analysis.py:6: in <module>
    from telescope_baseline.tools.pipeline.map_on_detector_builder import MapOnDetectorBuilder
/usr/local/lib/python3.9/site-packages/telescope_baseline-0.1.0-py3.9.egg/telescope_baseline/tools/pipeline/map_on_detector_builder.py:2: in <module>
    from photutils.detection import find_peaks
E   ModuleNotFoundError: No module named 'photutils'
_______________________________ ERROR collecting tests/test_tools/assert_timeout_test.py _______________________________
ImportError while importing test module '/opt/jasmine/tests/test_tools/assert_timeout_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_tools/assert_timeout_test.py:3: in <module>
    from test_tools.assert_timeout import assert_timeout
E   ModuleNotFoundError: No module named 'test_tools'
=============================================== short test summary info ================================================
ERROR slowtests/test_slow_test_example.py
ERROR tests/pipeline/test_astrometric_catalogue.py
ERROR tests/pipeline/test_detector_image.py
ERROR tests/pipeline/test_detector_image_builder.py
ERROR tests/pipeline/test_detector_image_catalogue.py
ERROR tests/pipeline/test_map_on_detector.py
ERROR tests/pipeline/test_map_on_detector_builder.py
ERROR tests/pipeline/test_map_on_the_sky.py
ERROR tests/pipeline/test_pipeline.py
ERROR tests/test_tools/assert_timeout_test.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 10 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================== 10 errors in 5.17s ==================================================
r

--- a/tests/pipeline/test_astrometric_catalogue.py
+++ b/tests/pipeline/test_astrometric_catalogue.py
@@ -1,12 +1,11 @@
 import csv
 import pytest
-import tempfile
 from astropy.coordinates import SkyCoord
 import astropy.units as u
 from telescope_baseline.tools.pipeline.astrometric_catalogue import AstrometricCatalogue
 from telescope_baseline.tools.pipeline.catalog_entry import CatalogueEntry
 from test_pipeline import get_tests_file_name
-
+from test_tools.test_file_utils import get_temp_file
 
 @pytest.fixture
 def csv_line():
@@ -16,12 +15,12 @@ def csv_line():
                                 distance=1.0 * u.pc, pm_l_cosb=10 * u.mas / u.yr, pm_b=10 * u.mas / u.yr),
                        12.5),
     ])
-    with tempfile.NamedTemporaryFile() as tmp_file:
-        f_name = tmp_file.name
-        a.save(f_name)
-        file = open(f_name, 'r', newline='')
-        f = csv.reader(file, delimiter=',')
-        return next(iter(f))
+    tmp_file = get_temp_file()
+    f_name = str(tmp_file)
+    a.save(f_name)
+    file = open(f_name, 'r', newline='')
+    f = csv.reader(file, delimiter=',')
+    return next(iter(f))
 
 
 def test_save1(csv_line):

--- a/tests/pipeline/test_map_on_detector.py
+++ b/tests/pipeline/test_map_on_detector.py
@@ -1,6 +1,5 @@
 import csv
 import pytest
-import tempfile
 from astropy.time import Time
 from astropy.wcs import WCS
 from telescope_baseline.tools.pipeline.map_on_detector import MapOnDetector
@@ -8,6 +7,7 @@ from telescope_baseline.tools.pipeline.map_on_the_sky_builder import MapOnTheSky
 from telescope_baseline.tools.pipeline.position2d import Position2D
 from telescope_baseline.tools.pipeline.position_on_detector import PositionOnDetector
 from test_pipeline import get_tests_file_name
+from test_tools.test_file_utils import get_temp_file
 
 
 @pytest.fixture
@@ -18,11 +18,12 @@ def mod():
 
 @pytest.fixture
 def csv_read(mod):
-    with tempfile.NamedTemporaryFile() as tmp_file:
-        mod.save(tmp_file.name)
-        file = open(tmp_file.name, 'r', newline='')
-        f = csv.reader(file, delimiter=',')
-        return next(iter(f))
+    tmp_file = get_temp_file()
+    file_name = str(tmp_file)
+    mod.save(file_name)
+    file = open(file_name, 'r', newline='')
+    f = csv.reader(file, delimiter=',')
+    return next(iter(f))
 
 
 def test_save1(csv_read):

--- a/tests/pipeline/test_map_on_the_sky.py
+++ b/tests/pipeline/test_map_on_the_sky.py
@@ -1,11 +1,11 @@
 import csv
 import pytest
-import tempfile
 from astropy.coordinates import SkyCoord
 from astropy.time import Time
 from telescope_baseline.tools.pipeline.map_on_the_sky import MapOnTheSky
 from telescope_baseline.tools.pipeline.position_on_the_sky import PositionOnTheSky
 from test_pipeline import get_tests_file_name
+from test_tools.test_file_utils import get_temp_file
 
 
 @pytest.fixture
@@ -17,20 +17,20 @@ def mos():
 
 
 def test_save(mos):
-    with tempfile.NamedTemporaryFile() as tmp_file:
-        f_name = tmp_file.name
-        mos.save(f_name)
-        file = open(f_name, 'r', newline='')
-        csv_line = csv.reader(file, delimiter=',')
-        for r in csv_line:
-            if len(r) == 1:
-                assert int(r[0]) == -1
-            else:
-                assert int(r[0]) == 1
-                assert abs(float(r[1]) - 4.649644189337132) < 0.001
-                assert abs(float(r[2]) + 0.5050315748856247) < 0.001
-                assert abs(float(r[3]) - 12.5) < 0.1
-                assert r[4] == '2000-01-01 00:00:00.000'
+    tmp_file = get_temp_file()
+    f_name = str(tmp_file)
+    mos.save(f_name)
+    file = open(f_name, 'r', newline='')
+    csv_line = csv.reader(file, delimiter=',')
+    for r in csv_line:
+        if len(r) == 1:
+            assert int(r[0]) == -1
+        else:
+            assert int(r[0]) == 1
+            assert abs(float(r[1]) - 4.649644189337132) < 0.001
+            assert abs(float(r[2]) + 0.5050315748856247) < 0.001
+            assert abs(float(r[3]) - 12.5) < 0.1
+            assert r[4] == '2000-01-01 00:00:00.000'
 
 
 @pytest.fixture

--- a/tests/test_tools/test_file_utils.py
+++ b/tests/test_tools/test_file_utils.py
@@ -1,5 +1,7 @@
 from pathlib import Path
 import sys
+import os
+import tempfile
 
 
 def get_test_file(file_path: str) -> Path:
@@ -8,3 +10,40 @@ def get_test_file(file_path: str) -> Path:
         if target_file.exists():
             return target_file
     raise FileNotFoundError()
+
+
+class _PathWrapper:
+    def __init__(self, origin:Path, delete_when_gc):
+        self.__origin = origin
+        self.__delete_when_gc = delete_when_gc
+
+    def __getattr__(self, name):
+        def func(*args, **kwargs):
+            return getattr(self.__origin, name)(*args, **kwargs)
+        return func
+
+    def __repr__(self):
+        return repr(self.__origin)
+
+    def __bytes__(self):
+        return bytes(self.__origin)
+
+    def __str__(self):
+        return str(self.__origin)
+
+    def __format__(self, form_spec):
+        return format(self.__origin, form_spec)
+
+    def __del__(self):
+        if self.__delete_when_gc:
+            self.__origin.unlink(missing_ok=True)
+
+
+def get_temp_file(delete_when_gc: bool = True) -> Path:
+    tmp_name = os.path.join(tempfile.gettempdir(), os.urandom(24).hex())
+    path = Path(tmp_name)
+    if path.exists():
+        return get_temp_file()
+    else:
+        path.touch()
+        return _PathWrapper(path, delete_when_gc)

--- a/tests/test_tools/test_file_utils.py
+++ b/tests/test_tools/test_file_utils.py
@@ -1,6 +1,10 @@
 from pathlib import Path
+import sys
 
 
-def get_test_file(file_path:str, test_root_dir:str = 'tests') -> Path:
-    project_root = Path.cwd()
-    return Path(project_root, test_root_dir, file_path)
+def get_test_file(file_path: str) -> Path:
+    for sys_path in sys.path:
+        target_file = Path(sys_path, file_path)
+        if target_file.exists():
+            return target_file
+    raise FileNotFoundError()

